packages/express/src/middleware.test.ts

```diff
@@ -0,0 +1,197 @@
+import Cipher from '@bearer/security'
+
+import { middleware, TWebhookHandlers } from './middleware'
+
+const SUCCESS_HANDLER = 'sponge-bob-integration-handler'
+const REJECTED_HANDLER = 'patrick-is-rejecting'
+const FAILING_HANDLER = 'patrick-is-failing'
+
+const webHookHandlers = {
+  [SUCCESS_HANDLER]: jest.fn((req: Request & { title: string }) => Promise.resolve(req.title)),
+  [REJECTED_HANDLER]: jest.fn((req: Request) => Promise.reject()),
+  [FAILING_HANDLER]: jest.fn((req: Request) => {
+    throw new Error('ok')
+  })
+} as TWebhookHandlers
+
+describe('Bearer middleware', () => {
+  beforeEach(() => {
+    Object.keys(webHookHandlers).map(key => {
+      // @ts-ignore
+      webHookHandlers[key].mockClear()
+    })
+  })
+
+  describe('middleware logic', () => {
+    const app = express()
+    app.use((req: Request & { title?: string }, _res, next) => {
+      req.title = 'Sponge bob is the king'
+      next()
+    })
+    app.use('/whatever/webhooks', middleware(webHookHandlers))
+
+    app.post('/*', (_req, res) => {
+      res.status(200).json({ name: 'john' })
+    })
+
+    app.get('/*', (_req, res) => {
+      res.status(200).json({ name: 'john' })
+    })
+
+    describe('existing handlers', () => {
+      it('calls handler with req object', async () => {
+        await request(app)
+          .post('/whatever/webhooks')
+          .set('BEARER-INTEGRATION-HANDLER', SUCCESS_HANDLER)
+
+        const mock = webHookHandlers[SUCCESS_HANDLER] as jest.Mock
+        // request is passed down
+        expect(mock.mock.calls[0][0]).toBeInstanceOf(IncomingMessage)
+        expect(mock.mock.calls[0][0].title).toBe('Sponge bob is the king')
+      })
+
+      it('with alias - calls handler with req object', async () => {
+        await request(app)
+          .post('/whatever/webhooks')
+          .set('BEARER-INTEGRATION-HANDLER', 'whatever')
+          .set('BEARER-INTEGRATION-ALIAS', SUCCESS_HANDLER)
+
+        const mock = webHookHandlers[SUCCESS_HANDLER] as jest.Mock
+        // request is passed down
+        expect(mock.mock.calls[0][0]).toBeInstanceOf(IncomingMessage)
+        expect(mock.mock.calls[0][0].title).toBe('Sponge bob is the king')
+      })
+
+      it('returns correct success payload', async () => {
+        const response = await request(app)
+          .post('/whatever/webhooks')
+          .set('BEARER-INTEGRATION-HANDLER', SUCCESS_HANDLER)
+          .expect('Content-Type', /json/)
+          .expect(200)
+
+        expect(webHookHandlers[SUCCESS_HANDLER]).toHaveBeenCalled()
+        expect(response.body).toMatchObject({ ack: 'ok' })
+      })
+
+      it('perform promise even if rejecting', async () => {
+        const response = await request(app)
+          .post('/whatever/webhooks')
+          .set('BEARER-INTEGRATION-HANDLER', REJECTED_HANDLER)
+          .expect('Content-Type', /json/)
+          .expect(422, {
+            error: { name: 'Bearer:WebhookHandlerRejection', message: 'patrick-is-rejecting' }
```

Comment:
Is this right?
```suggestion
            error: { name: 'Bearer:WebhookHandlerRejection', message: REJECTED_HANDLER }
```

Generally in here if you could use constants more over magic strings it would make things easier to follow

Reply:
To be honest I try to avoid re-use variable when comes to expectations. 
If a variable is wrong then all your tests fail without you knowing. while copy/pasting does not prevent you to copy 💩 it reduces risk IMO.